addons/udes_stock/views/stock_picking.xml

```diff
@@ -158,6 +158,30 @@
                 <attribute name="attrs">{'readonly': [('state', '!=', 'draft')],  'required': [('picking_type_code', '=', 'incoming')]}</attribute>
             </xpath>
 
+            <xpath expr="//field[@name='date_done']" position="attributes">
+                <attribute name="attrs">{'invisible': [('state', 'not in', ('done', 'cancel'))]}</attribute>
+                <attribute name="string">Date of Transfer</attribute>
+            </xpath>
+
+        </field>
+    </record>
+
+
+    <record id="vpicktree" model="ir.ui.view">
+        <field name="name">stock.picking.tree.udes</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.vpicktree"/>
+        <field name="arch" type="xml">
+
+            <field name="scheduled_date" position="attributes">
+                <attribute name="widget">datetime</attribute>
+                <attribute name="attrs"></attribute>
+            </field>
+
+            <field name="scheduled_date" position="after">
```

Comment:
I don't have a problem personally with hooking into fields this way, but we've tended to use `xpath` as standard.

Will check tomorrow because I can't remember if we had any policy around this tbh, but this can probably be ignored. :)

Reply:
Ok it's fine , know that odoo itself uses this way when its can be found easily example not inside pages etc.
But happy to change with xpath if we have that standard

Reply:
Will resolve, couldn't find anything.

Also as it is a tree view we're only ever going to have a `tree` element with `field` and `button` elements inside. So we don't have the potential like form views where you can have fields from one2many, many2many etc.